### PR TITLE
Namespace functions and parameters

### PR DESCRIPTION
**Pull request by zmully**
I'm not particularly happy with the tests, but I wanted to get this out there for another set of eyes to take a look at them since I'm out of ideas how to work around the path issues during testing. 

/cc @ianshward

### CODE REVIEW

File: lib/lambda-cfn.js [hunk 1]
```diff
@@ -24,6 +24,28 @@ lambdaCfn.lambdaSnsUser = lambdaSnsUser;
 lambdaCfn.lambdaSnsUserAccessKey = lambdaSnsUserAccessKey;
 lambdaCfn.outputs = outputs;
 lambdaCfn.load = load;
+lambdaCfn.getEnv = getEnv;
+
+function stripHyphens(r) {
+    return r.replace(/-/g, '');
+}
+
+var packages;
```
Comment (ianshward) [inline lib/lambda-cfn.js]: @zmully any reason not to put these lines inside the `namespace` function itself?

File: lib/lambda-cfn.js [hunk 2]
```diff
@@ -24,6 +24,28 @@ lambdaCfn.lambdaSnsUser = lambdaSnsUser;
...
+if (process.env.NODE_ENV == 'test') {
+  packages = require(path.join(root,'/test/package.json'));
+} else {
+  packages = require(path.join(root,'package.json'));
+}
+
+function namespace(name,path) {
```
Comment (ianshward) [inline lib/lambda-cfn.js]: @zmully you _might_ be able to use module.children - globally available - here if you're looking for a way to do this w/o reaching out to package.json.

Comment (zmully) [inline lib/lambda-cfn.js]: @ianshward thanks for pointing that out, looking at that object, I think I can crawl it to extract the right information and avoid this ugly hacking about with the package.json.

**Comment by zmully** [comment]
Talk about cascading issues. Using `module.children` exposed a problem with the parameter namespacing where they were only being namespaced `ruleParameter` and not `repoRuleParameter`. 

Since the function name is need by the lambda to find its environment parameters from Streambot and is not easily found by the function itself, I ended choosing to iterate through the `process.env` to match the known function parameters against. Another strategy would be to pull the lambda function name from the Lambda context object and parse that.

I ran into a testing problem that I couldn't solve where subsequent tests would get stale rule configuration information if the same rule definition was reused between tests. While I'd love to figure this out, I burnt too much time on this already, so I simply changed the lambda-rules.test.js to use another test rule instead of the assumeRole.js rule that was already there.

**Comment by zmully** [comment]
I'm going to merge this work so that the new CFN support can be built with namespace support. I'll open another PR to remove the lambda-rules.js binary later.

**Pull request merged**